### DISCUSSION TITLE
Add frame id in Detection class

### DISCUSSION THREAD

**Discussion by hardikdava** [roboflow/supervision#228] [Ideas]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Description

Curent `Detection` class is missing an information of `frame id`. Generally working with multiple object tracking it is required to have information of detections associated with frame of a sequence or video frame. 

### Use case

Use cases would be as following:

- Drawing trace of a tracker
- Tracker evaluation
- Tracker dataset import/export
- Correcting tracker mistakes

### Additional

Implementation possibility:
- Add an optional argument in existing `Detection` class

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by SkalskiP** [comment]
Hi @hardikdava 👋🏻 I was thinking about adding `frame_id` but eventually decided not to do it. Do you really think we need it? Couldn't we just load the tracker dataset as `Dict[str, Detection]` where the key is `frame_id`?

**Comment by hardikdava** [comment]
Hi @SkalskiP  👋🏻 , we can introduce `sv.TrackingDataset` as a new feature just like another dataset format. It can be in a format what you suggested. Later, there is also a possibility to introduce `TrackerEvaluation` What do you think about it?

**Comment by hardikdava** [comment]
Hi, @SkalskiP I have integrate [MOT-Dataset](https://motchallenge.net/data/MOT20/) with `DetectionDataset` but `supervision` can also introduce as `sv.TrackingDataset` due to used for different computer vision task. Let me know if you want to integrate it. I am willing to submit a new PR for this.

**Comment by SkalskiP** [comment]
I'll convert that into the discussion for now and add it to `Ideas`

**Comment by hardikdava** [comment]
@SkalskiP I have started working on it. It will be minor but important feature after `Tracker` support. You can count this on release `0.13.0`.
